ui/plugins/pinboard.js

```diff
@@ -101,6 +101,15 @@ treeherder.controller('PinboardCtrl', [
             }
         };
 
+        $scope.retriggerAllPinnedJobs = function() {
+            var jobs = [];
+            // pushing pinned jobs to a list.
+            for (var job in $scope.pinnedJobs) {
+                jobs.push($scope.pinnedJobs[job]);
+            }
+            $scope.retriggerJob(jobs);
```

Comment:
Maybe I'm missing some context, but it looks like you can pass $scope.pinnedJobs  to retriggerJob directly.